databricks_cli/stack/api.py

```diff
@@ -0,0 +1,246 @@
+
+    def _load_deploy_metadata(self, stack_path, save_path=None):
+        parsed_conf = {}
+        default_status_path = self._generate_stack_status_path(stack_path)
+        try:
+            if os.path.exists(default_status_path):
+                with open(default_status_path, 'r') as f:
+                    parsed_conf = json.load(f)
+                click.echo("Using deployment status file at %s" % default_status_path)
+            elif save_path and os.path.exists(save_path):
+                with open(save_path, 'r') as f:
+                    parsed_conf = json.load(f)
+                click.echo("Using deployment status file at %s" % stack_path)
+        except ValueError:
+            pass
+
+        if STACK_RESOURCES in parsed_conf:
+            self.deployed_resource_config = parsed_conf[STACK_RESOURCES]
+        if STACK_DEPLOYED in parsed_conf:
+            self.deployed_resources = {resource[RESOURCE_ID]: resource for resource in
+                                       parsed_conf[STACK_DEPLOYED]}
+
+        return parsed_conf
+
+    def _get_deployed_resource(self, resource_id, resource_type):
+        """
+        Returns the databricks physical ID of a resource with RESOURCE_ID and RESOURCE_TYPE
+
+        :param resource_id: Internal stack identifier of resource
+        :param resource_type: Resource type of stack resource
+        :return: JSON object of Physical ID of resource on databricks
+        """
+        if not self.deployed_resources:
+            return None
+        if resource_id in self.deployed_resources:
+            deployed_resource = self.deployed_resources[resource_id]
+            deployed_resource_type = deployed_resource[RESOURCE_TYPE]
+            deployed_physical_id = deployed_resource[RESOURCE_PHYSICAL_ID]
+            if resource_type != deployed_resource_type:
+                click.echo("Resource %s is not of type %s", (resource_id, resource_type))
+                return None
+            return deployed_physical_id
+        return None
+
+    def _store_deploy_metadata(self, stack_path, data, custom_path=None):
+        stack_status_filepath = self._generate_stack_status_path(stack_path)
+        stack_file_folder = os.path.dirname(stack_status_filepath)
+        if not os.path.exists(stack_file_folder):
+            os.makedirs(stack_file_folder)
+        with open(stack_status_filepath, 'w+') as f:
+            click.echo('Storing deploy status metadata to %s' % stack_status_filepath)
+            json.dump(data, f, indent=2)
+
+        if custom_path:
+            custom_path_folder = os.path.dirname(custom_path)
+            if not os.path.exists(custom_path_folder):
+                os.makedirs(custom_path_folder)
+            with open(custom_path, 'w+') as f:
+                click.echo('Storing deploy status metadata to %s' % os.path.abspath(custom_path))
+                json.dump(data, f, indent=2)
+
+    def deploy_job(self, resource_id, job_settings, physical_id=None):
```

Comment:
`physical_id` => `job_id` so it is more specific?

Reply:
Hm yeah that will actually work